### PR TITLE
InaSAFE Headless cumulative PR

### PR DESCRIPTION
**Pull request by lucernae**
Contains cumulative changes from merging InaSAFE 3.4 to InaSAFE Headless packages.

### CODE REVIEW

File: bin/inasafe.py [hunk 1]
```diff
@@ -32,11 +33,15 @@
     QgsMapLayerRegistry,
     QgsCoordinateReferenceSystem)
 
+
+from safe.test.utilities import get_qgis_app
+# make sure this line executes first
+QGIS_APP, CANVAS, IFACE, PARENT = get_qgis_app()
```
Comment (Gustry) [inline bin/inasafe.py]: Do you use IFACE and CANVAS outside of our test suite ? (I mean in production)

Comment (timlinux) [inline bin/inasafe.py]: Yeah we should probably not be importing test code into production. Conceivably we could be stripping test code out of production shipped packages.

**Comment by timlinux** [comment]
LGTM thanks @lucernae

File: bin/inasafe.py (hunk 1 above)
Comment (lucernae) [inline bin/inasafe.py]: I didn't use the variable specifically, but get_qgis_app needs to be called first to initialize QPaint and other Qt related object for the composer to works.
Any solution besides this is highly appreciated.

**Comment by timlinux** [comment]
Can you merge this please @lucernae ? We are supposed to be in feature freeze from 1 July...

**Comment by timlinux** [comment]
LGRTM :-)

**Pull request merged**